service/controller/appcatalog/resource/appcatalogentry/create.go

```diff
@@ -140,34 +139,25 @@ func (r *Resource) newAppCatalogEntries(ctx context.Context, cr v1alpha1.AppCata
 		for _, entry := range entries {
 			name := fmt.Sprintf("%s-%s-%s", cr.Name, entry.Name, entry.Version)
 
-			var metadata []byte
+			var rawMetadata []byte
 			{
-				metadata, err = r.getMetadata(ctx, key.AppCatalogStorageURL(cr), entry.Name, entry.Version)
+				rawMetadata, err = r.getMetadata(ctx, key.AppCatalogStorageURL(cr), entry.Name, entry.Version)
 				if err != nil {
 					r.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("failed to get metadata for entry %#q in catalog %#q", entry.Name, cr.Name), "stack", fmt.Sprintf("%#v", err))
 					continue
 				}
 			}
 
-			var restrictions *v1alpha1.AppCatalogEntrySpecRestrictions
+			var metadata *metadata
 			{
-				if metadata != nil {
-					restrictions, err = parseRestrictions(metadata)
+				if rawMetadata != nil {
+					metadata, err = parseMetadata(rawMetadata)
 					if err != nil {
 						return nil, microerror.Mask(err)
 					}
 				}
 			}
 
-			createdTime, err := parseTime(entry.Created)
```

Comment:
Why not utilize `dateCreated` from metadata files directly? No need to use the entry date from the catalog.

Reply:
Skipping this arguing for simplicity.

Reply:
As we discussed in the team channel, resource would fall back to the date in the entry if no `dateCreated` is provided by meta.yaml.